pkg/actions/project.go

```diff
@@ -135,3 +139,106 @@ func ProjectRemoveConnection(c *cli.Context) {
 	fmt.Println(string(response))
 	os.Exit(0)
 }
+
+// ProjectList : Print the list of projects to the terminal
+func ProjectList(c *cli.Context) {
+	conID := strings.TrimSpace(strings.ToLower(c.String("conid")))
+
+	conInfo, conInfoErr := connections.GetConnectionByID(conID)
+	if conInfoErr != nil {
+		HandleConnectionError(conInfoErr)
+		os.Exit(1)
+	}
+
+	conURL, conErr := config.PFEOriginFromConnection(conInfo)
+	if conErr != nil {
+		HandleConfigError(conErr)
+		os.Exit(1)
+	}
+
+	projects, getAllErr := project.GetAll(http.DefaultClient, conInfo, conURL)
+	if getAllErr != nil {
+		HandleProjectError(getAllErr)
+		os.Exit(1)
+	}
+	if printAsJSON {
+		json, _ := json.Marshal(projects)
+		fmt.Println(string(json))
+	} else {
+		if len(projects) == 0 {
+			fmt.Println("No projects bound to Codewind")
+		} else {
+			w := new(tabwriter.Writer)
+			w.Init(os.Stdout, 0, 8, 0, '\t', 0)
+			fmt.Fprintln(w, "PROJECT ID \tNAME \tLANGUAGE \tAPP STATUS \tLOCATION ON DISK")
+			for _, project := range projects {
+				appStatus := strings.Title(project.AppStatus)
+				fmt.Fprintln(w, project.ProjectID+"\t"+project.Name+"\t"+project.Language+"\t"+appStatus+"\t"+project.LocationOnDisk)
+			}
+			fmt.Fprintln(w)
+			w.Flush()
+		}
+	}
+	os.Exit(0)
+}
+
+// ProjectGet : Prints information about a given project using its ID
+func ProjectGet(c *cli.Context) {
+	conID := strings.TrimSpace(strings.ToLower(c.String("conid")))
+	projectID := strings.TrimSpace(strings.ToLower(c.String("id")))
+	projectName := c.String("name")
+
+	if projectID == "" && projectName == "" {
+		fmt.Println("Error: Must specify either project ID (--id) or project name (--name)")
+		os.Exit(1)
+	}
+
+	if projectID != "" && (conID == "local" || conID == "") {
+		newConID, conIDErr := project.GetConnectionID(projectID)
+		if conIDErr != nil {
+			HandleProjectError(conIDErr)
+			os.Exit(1)
+		}
+		conID = newConID
+	}
+
+	conInfo, conInfoErr := connections.GetConnectionByID(conID)
+	if conInfoErr != nil {
+		HandleConnectionError(conInfoErr)
+		os.Exit(1)
+	}
+
+	conURL, conErr := config.PFEOriginFromConnection(conInfo)
+	if conErr != nil {
+		HandleConfigError(conErr)
+		os.Exit(1)
+	}
+
+	if projectID == "" && projectName != "" {
+		newProjectID, projectNameErr := project.GetProjectIDFromName(http.DefaultClient, conInfo, conURL, projectName)
```

Comment:
Could have a project.getProjectFromName and a project.getProjectFromID

Reply:
Done this